src/generator/named-types.ts

```diff
@@ -42,7 +51,49 @@ export function dropIPrefixInCustomTypes(modules: Module[]): void {
     });
 }
 
-export function fetchNamedTypes(modules: Module[]): NamedTypesResult {
+export function parseTarget(modules: Module[]): ParsedTarget {
+  const namedTypes = fetchNamedTypes(modules);
+
+  return {
+    modules: modules.map((module) => ({
+      ...module,
+      associatedTypes: namedTypes.associatedTypes[module.name] ?? [],
+    })),
+    sharedTypes: namedTypes.sharedTypes,
+  };
+}
+
+export function extractSharedTypes(targets: ParsedTarget[]): NamedTypeInfo[] {
+  const typeTargetsMap: Record<string, [NamedTypeInfo, Set<ParsedTarget>]> = {};
```

Comment:
Looks like we're having shared types crossing modules and shared type crossing targets? They're both named sharedTypes which could be confusing. Could we name them explicitly?

Reply:
That's right. I think the name is fine since even though they share the same name, they are attached to different interfaces, and can be easily inferred from the context.

Reply:
Agreed on that part. But like for this method, I have to read the code to understand which `sharedTypes` to extract and what this method does.

Reply:
I thought the parameter type of the function signature was obvious enough. Let me change the name then.